### ISSUE TITLE
Streams not working

### ISSUE THREAD

**Issue by digitalkaoz** [streamich/unionfs#159]
hey @streamich 

this wont work as expected, what wrong here. it searched the PRs and Issues, and Streams should be resolved somehow (it probably uses all your fs related packages):

```js
//patching everything
const { patchRequire } = require("fs-monkey");
const { ufs } = require("unionfs");
const { vol } = require("memfs");
const mock = require("mock-require");
const realFs = require("fs");

vol.fromJSON(
  {
    "./dir": null,
    "./file.json": "{\"foo\":\"bar\"}"
  },
  __dirname
);

ufs.use(vol).use(realFs);

patchRequire(ufs);
mock("fs", ufs);


//the code inside the app
const fs = require("fs");

// a file from the real FS with or without UFS
console.log("realfs buffer", fs.readFileSync(__dirname + "/package.json")); //buffer
console.log("realfs ufs stream", fs.createReadStream(__dirname + "/package.json")); //undefined
console.log("realfs stream", realFs.createReadStream(__dirname + "/package.json")); //stream
console.log(require(__dirname + "/package.json")); //works

// a file from the memory fs with or without UFS
console.log("memfs buffer", fs.readFileSync(__dirname + "/file.json")); //buffer
console.log("memfs stream", vol.createReadStream(__dirname + "/file.json")); //stream
console.log(require(__dirname + "/file.json")); //works
console.log("memfs stream through ufs", fs.createReadStream(__dirname + "/file.json")); //undefined + ENOENT error
```

it would be awesome if you can help me?

If im understanding the code correctly it should work (its testing all underlying `fs`'s for `createReadStream` (continues if one errors, that might be the problem?)

**Comment by digitalkaoz** [comment]
i created a try to fix branch here: https://github.com/digitalkaoz/unionfs/tree/fix/streams

[x] failing test case
[ ] fix for readable streams
[ ] fix for writable streams
[ ] pick correct stream from the underlying fs

**Comment by streamich** [comment]
Streams are not well supported by `unionfs`, to make streams work properly somebody needs to port [this code](https://github.com/streamich/memfs/blob/master/src/volume.ts#L2101-L2422) and adapting it to work with `unionfs`.

**Comment by streamich** [comment]
And this code:

https://github.com/streamich/memfs/blob/master/src/volume.ts#L2006-L2012

**Comment by digitalkaoz** [comment]
@streamich i might found a simpler solution (that fixes the above test scenario), see #160
